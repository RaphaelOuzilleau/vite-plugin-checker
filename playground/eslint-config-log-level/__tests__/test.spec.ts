// import stringify from 'fast-json-stable-stringify'
import { describe, expect, it } from 'vitest'

import {
  resetReceivedLog,
  sleepForEdit,
  sleepForServerReady,
  stripedLog,
  editFile,
  isServe,
} from '../../testUtils'
import stringify from 'fast-json-stable-stringify'

describe('eslint-config-log-level', () => {
  describe.runIf(isServe)('serve', () => {
    it('get initial error and subsequent error', async () => {
      let diagnostics: any
      await sleepForServerReady()
      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()

      console.log('-- edit error file --')
      resetReceivedLog()
      editFile('src/main.ts', (code) => code.replace(`'Hello'`, `'Hello~'`))
      await sleepForEdit()
      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()
    })
  })
})
