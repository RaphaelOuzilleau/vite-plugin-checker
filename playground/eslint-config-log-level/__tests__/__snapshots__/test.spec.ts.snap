// Vitest Snapshot v1

exports[`eslint-config-log-level > serve > get initial error and subsequent error 1`] = `undefined`;

exports[`eslint-config-log-level > serve > get initial error and subsequent error 2`] = `
" ERROR(ESLint)  Unexpected var, use let or const instead. (no-var)
 FILE  <PROJECT_ROOT>/playground-temp/eslint-config-log-level/src/main.ts:3:1

    1 | import { text } from './text'
    2 |
  > 3 | var hello = 'Hello'
      | ^^^^^^^^^^^^^^^^^^^
    4 |
    5 | const rootDom = document.querySelector('#root')!
    6 | rootDom.innerHTML = hello + text
 WARNING(ESLint)  Forbidden non-null assertion. (@typescript-eslint/no-non-null-assertion)
 FILE  <PROJECT_ROOT>/playground-temp/eslint-config-log-level/src/main.ts:5:17

    3 | var hello = 'Hello'
    4 |
  > 5 | const rootDom = document.querySelector('#root')!
      |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    6 | rootDom.innerHTML = hello + text
    7 |
    8 | export {}
[ESLint] Found 1 error and 1 warning"
`;

exports[`eslint-config-log-level > serve > get initial error and subsequent error 3`] = `undefined`;

exports[`eslint-config-log-level > serve > get initial error and subsequent error 4`] = `
" ERROR(ESLint)  Unexpected var, use let or const instead. (no-var)
 FILE  <PROJECT_ROOT>/playground-temp/eslint-config-log-level/src/main.ts:3:1

    1 | import { text } from './text'
    2 |
  > 3 | var hello = 'Hello~'
      | ^^^^^^^^^^^^^^^^^^^^
    4 |
    5 | const rootDom = document.querySelector('#root')!
    6 | rootDom.innerHTML = hello + text
 WARNING(ESLint)  Forbidden non-null assertion. (@typescript-eslint/no-non-null-assertion)
 FILE  <PROJECT_ROOT>/playground-temp/eslint-config-log-level/src/main.ts:5:17

    3 | var hello = 'Hello~'
    4 |
  > 5 | const rootDom = document.querySelector('#root')!
      |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    6 | rootDom.innerHTML = hello + text
    7 |
    8 | export {}
[ESLint] Found 1 error and 1 warning"
`;
